src/idpi/metadata.py

```diff
@@ -0,0 +1,39 @@
+"""Manage GRIB metadata."""
+
+# Standard library
+import io
+import typing
+
+# Third-party
+import earthkit.data as ekd
+from earthkit.data.writers import write
+
+
+def override(message: bytes, **kwargs: typing.Any) -> dict[str, typing.Any]:
+    """Override GRIB metadata contained in message.
+
+    Parameters
+    ----------
+    message : bytes
+        Byte string of the input GRIB message
+    kwargs : Any
+        Keyword arguments forwarded to earthkit-data GribMetadata override method
+
+    Returns
+    -------
+    dict[str, Any]
+        Updated message byte string along with the geography and parameter namespaces
+
+    """
+    stream = io.BytesIO(message)
+    [grib_field] = ekd.from_source("stream", stream)
```

Comment:
I think that this works, as long as you do not have to use another product definition template to code the result.